helix-term/src/ui/mod.rs

```diff
@@ -208,6 +210,34 @@ pub mod completers {
         names
     }
 
+    pub fn setting(input: &str) -> Vec<Completion> {
+        static KEYS: Lazy<Vec<Completion>> = Lazy::new(|| {
+            use serde_json::{Map, Value};
+            serde_json::de::from_slice::<Map<String, Value>>(
+                &serde_json::ser::to_vec(&Config::default()).unwrap(),
+            )
+            .unwrap()
+            .keys()
+            .map(|key| ((0..), Cow::from(key.to_string())))
+            .collect()
+        });
+
+        let matcher = Matcher::default();
+
+        let mut matches: Vec<_> = KEYS
+            .iter()
+            .filter_map(|(_range, name)| {
+                matcher.fuzzy_match(name, input).map(|score| (name, score))
+            })
+            .collect();
+
+        matches.sort_unstable_by_key(|(_file, score)| Reverse(*score));
+        matches
+            .into_iter()
+            .map(|(name, _)| ((0..), name.clone()))
```

Comment:
We should be able to use `name.into()` here to construct a `Cow<str>` here since the lifetime of the names should be `'static`

Reply:
Huuum no I don't think I can because I only have a ref on the name since it comes from my static var `KEYS`, I do not own the `Cow` directly and thus I need to clone it before returning it :thinking:

Reply:
Yeah, now I noticed, you're constructing a `(range, name)`, then throwing away the range and mapping it again later. Here's a diff that worked for me that should be more efficient

```diff
diff --git i/helix-term/src/ui/mod.rs w/helix-term/src/ui/mod.rs
index 6d0e45b8..9e096311 100644
--- i/helix-term/src/ui/mod.rs
+++ w/helix-term/src/ui/mod.rs
@@ -211,13 +211,13 @@ pub mod completers {
     }
 
     pub fn setting(input: &str) -> Vec<Completion> {
-        static KEYS: Lazy<Vec<Completion>> = Lazy::new(|| {
+        static KEYS: Lazy<Vec<String>> = Lazy::new(|| {
             serde_json::to_value(Config::default())
                 .unwrap()
                 .as_object()
                 .unwrap()
                 .keys()
-                .map(|key| ((0..), Cow::from(key.to_string())))
+                .cloned()
                 .collect()
         });
 
@@ -225,15 +225,13 @@ pub mod completers {
 
         let mut matches: Vec<_> = KEYS
             .iter()
-            .filter_map(|(_range, name)| {
-                matcher.fuzzy_match(name, input).map(|score| (name, score))
-            })
+            .filter_map(|name| matcher.fuzzy_match(name, input).map(|score| (name, score)))
             .collect();
 
         matches.sort_unstable_by_key(|(_file, score)| Reverse(*score));
         matches
             .into_iter()
-            .map(|(name, _)| ((0..), name.clone()))
+            .map(|(name, _)| ((0..), name.into()))
             .collect()
     }
```

Reply:
That's smart, I applied your patch :+1: